### ISSUE TITLE
Add behaviour verification to Test2::Mock objects

### ISSUE THREAD

**Issue by MCRayRay** [Test-More/Test2-Suite#158]
# Summary

This is a feature request for allowing calls to a Test2::Mock object to be stored/returned. This is intended to be used by test scripts using behaviour verification to check the order and content of calls to a mocked object during a test run. What I have in mind is something similar to the implementation of [Test::MockObject](https://metacpan.org/pod/Test::MockObject).

# Initial Proposal
```perl
$mock->track('foo' [, sub{...} ]);

my $mock = mock obj => (
    track => [ foo => sub { ... } ],
); 
```

Records calls made to method 'foo', where the sub is optional if you want to
track the calls to a method defined another way.

```perl
my $calls = $mock->calls('foo');
```

Returns calls made via method 'foo' on a mocked object.

**Comment by robkinyon** [comment]
I'd like to assist on this, possibly providing a PR. I'm stuck on a design and I'm wondering what you think, @exodist. Some thoughts:

The simplest solution would be to copy what Test::MockObject does at https://github.com/chromatic/Test-MockObject/blob/master/lib/Test/MockObject.pm#L374. For interface compatibility, we'd add all the functions from lines 149-211. That seems to be a good set of functionality and it's well-understood.

The problem, on its face, is this ignores the idea of override and restore. However, I think that's okay. Most of the time I've seen the _calls() functionality used is when you create a mock that you hand to some other object. You then invoke a method on the second object and now you want to see what that object did with your mock. So, no-one is going to override/restore in the middle of a call-chain that they care about.

If you're happy with this design, I can get a PR to you in the next couple days.

**Comment by exodist** [comment]
*that is breaking compat with the other mock tool. Please do not break compat with the current Test2::Mock.

**Comment by jjatria** [comment]
This was implemented and merged in #184. It looks to me like this can be closed.
